pkg/handlers/jsonl.go

```diff
@@ -185,10 +185,16 @@ func handleJSONLObject(f *appendable.IndexFile, r []byte, dec *json.Decoder, pat
 					return fmt.Errorf("unexpected token '%v'", value)
 				}
 			case nil:
+				fmt.Printf("\n\ninserting data with offset: %v\n", data.Offset)
 				// nil values are a bit of a degenerate case, we are essentially using the btree
 				// as a set. we store the value as an empty byte slice.
-				if err := page.BPTree(r).Insert(btree.ReferencedValue{Value: []byte{}}, data); err != nil {
-					return fmt.Errorf("failed to insert into b+tree: %w", err)
+				if err := page.BPTree(r).Insert(btree.ReferencedValue{
+					Value:       []byte{},
+					DataPointer: data,
```

Comment:
The `csv_test`, `jsonl_test`, and the `index-file` equality tests are passing with one specific exception.

The case that fails is when we have a null column, that is when:
```json
{ "header": null }
{ "header": null }
{ "header": null }
```. 

The above approach seems to fix that test case, but I was wondering if it was intentional to zero out the `DataPointer` attribute in `ReferencedValue`

Note when we compare bytes, the `key` offset and length are always zeroed.